### ISSUE TITLE
Using InnoDB engine instead of MyISAM

### ISSUE THREAD

**Issue by jonthekiller** [ManiaControl/ManiaControl#263]
During the ZrT TM Cup, I worked on ManiaControl optimizations with 30 servers with 100 players linked to the same database. I already reported some optimizations for the Local Records Plugin but I also made a change with the Database Engine for all ManiaControl tables.

By default, they use MyISAM, but I decided to switch to InnoDB to increase performance that is vital with multi-servers on the same database. It makes a big changes with table locks, especially with Local Records Plugin where all servers can edit the table at the same time.

For simple and small servers, the impact is null, but for large and multi-servers the change is critical.

**Comment by axelalex2** [comment]
As far as i have read it up, both storage types have their fair use, and when introducing a new Database Backend, we might want to reevaluate where to use which storage type specifically, to get the absolute maximum performance out of every ManiaControl-class.

Alternatively, we could insert developer options in the configuration file, but this would only affect new installations, and might also divert from the simplicity of ManiaControl @kremsy envisions.

**Comment by kremsy** [comment]
Yes I agree, it has to be evaluated which performs best, but generally using Innodb could be an consideration. It depends how much is written / readen in database. But actually I think the Trackmania records stuff has most DB access, so it could be optimized for that.

**Comment by jonthekiller** [comment]
We have more selects. You can see in this graphs for 1 month with optimized LocalRecords plugin and InnoDB on 30 servers max.

![image](https://user-images.githubusercontent.com/5202324/89119130-074a6180-d4ac-11ea-9f6b-fb10c93854e9.png)
